sigstore-java/src/main/java/dev/sigstore/bundle/BundleReader.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2024 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.bundle;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.JsonFormat;
+import dev.sigstore.proto.ProtoMutators;
+import dev.sigstore.proto.common.v1.HashAlgorithm;
+import dev.sigstore.rekor.client.ImmutableInclusionProof;
+import dev.sigstore.rekor.client.ImmutableRekorEntry;
+import dev.sigstore.rekor.client.ImmutableVerification;
+import java.io.IOException;
+import java.io.Reader;
+import java.security.cert.CertPath;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.bouncycastle.util.encoders.Hex;
+
+public class BundleReader {
```

Comment:
just curious why a public class with one package protected method?

Reply:
too much refactoring, not enough self review